tests/e2e/tracer/test_tracer.py

```diff
@@ -0,0 +1,46 @@
+import datetime
+import json
+import uuid
+
+import boto3
+import pytest
+
+from .. import conftest
+from ..utils import helpers
+
+
+@pytest.fixture(scope="module")
+def config():
+    return {
+        "parameters": {"tracing": "ACTIVE"},
+        "environment_variables": {"ANNOTATION_KEY": f"e2e-tracer-{uuid.uuid4()}", "ANNOTATION_VALUE": "stored"},
+    }
+
+
+@pytest.mark.e2e
+def test_basic_lambda_trace_visible(execute_lambda: conftest.LambdaExecution, config: conftest.LambdaConfig):
+    lambda_arn = execute_lambda["arns"]["basichandlerarn"]
+    start_date = execute_lambda["execution_time"]
+    end_date = start_date + datetime.timedelta(minutes=5)
+
+    trace = helpers.get_traces(
+        start_date=start_date,
+        end_date=end_date,
+        lambda_function_name=lambda_arn.split(":")[-1],
+        xray_client=boto3.client("xray"),
+    )
+
+    for segment in trace["Traces"][0]["Segments"]:
```

Comment:
this might break if we add an additional test. What could be done here is to add a specific annotation per handler and fetch traces that contain that instead using filter expression.

Reply:
The idea was to fetch all new traces once and then manipulate/search them locally to avoid multiple calls to x-ray. Maybe we could simply iterate over downloaded traces and combine results in the list?